include/boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp

```diff
@@ -289,92 +266,69 @@ struct enriched_map_default_include_policy
     {
-        // Add all (non discarded) operations on this ring
-        // Blocked operations or uu on clusters (for intersection)
-        // should be included, to block potential paths in clusters
-        turn_type const& turn = *it;
-        if (turn.discarded)
+        if (!turn.discarded)
         {
-            continue;
-        }
-
-        std::size_t op_index = 0;
-        for (typename boost::range_iterator<container_type const>::type
-                op_it = boost::begin(turn.operations);
-            op_it != boost::end(turn.operations);
-            ++op_it, ++op_index)
-        {
-            if (include_policy.include(op_it->operation))
+            for_each_with_index(turn.operations, [&](std::size_t op_index, auto const& op)
             {
-                ring_identifier const ring_id
-                    (
-                        op_it->seg_id.source_index,
-                        op_it->seg_id.multi_index,
-                        op_it->seg_id.ring_index
-                    );
-                mapped_vector[ring_id].push_back
-                    (
-                        indexed_type(index, op_index, *op_it,
-                            it->operations[1 - op_index].seg_id)
-                    );
-            }
+                if (include_policy.include(op.operation))
+                {
+                    ring_identifier const ring_id
+                        (
+                            op.seg_id.source_index,
+                            op.seg_id.multi_index,
+                            op.seg_id.ring_index
+                        );
+                    mapped_vector[ring_id].emplace_back
+                        (
+                            index, op_index, op, turn.operations[1 - op_index].seg_id
+                        );
+                }
+            });
         }
-    }
-}
-
-template <typename Point1, typename Point2>
-inline typename geometry::coordinate_type<Point1>::type
-        distance_measure(Point1 const& a, Point2 const& b)
-{
-    // TODO: use comparable distance for point-point instead - but that
-    // causes currently cycling include problems
-    typedef typename geometry::coordinate_type<Point1>::type ctype;
-    ctype const dx = get<0>(a) - get<0>(b);
-    ctype const dy = get<1>(a) - get<1>(b);
-    return dx * dx + dy * dy;
+    });
 }
 
 template <typename Turns>
 inline void calculate_remaining_distance(Turns& turns)
 {
-    using turn_type = typename boost::range_value<Turns>::type;
-    using op_type = typename turn_type::turn_operation_type;
-
-    typename op_type::comparable_distance_type const zero_distance = 0;
-
-    for (auto it = boost::begin(turns); it != boost::end(turns); ++it)
+    for (auto& turn : turns)
     {
-        turn_type& turn = *it;
+        using ctype = typename geometry::coordinate_type<decltype(turn.point)>::type;
+        auto distance_measure = [](auto const& a, auto const& b) -> ctype
```

Comment:
This is not handled correctly by gcc up to version 7
For that reason I will undo this change to lambda